plugin/src/create-schema-customization.ts

```diff
@@ -0,0 +1,225 @@
+import { CreateSchemaCustomizationArgs } from "gatsby";
+
+export function createSchemaCustomization(
+  { actions, schema }: CreateSchemaCustomizationArgs,
+  pluginOptions: ShopifyPluginOptions
+) {
+  const includeCollections =
+    pluginOptions.shopifyConnections &&
+    pluginOptions.shopifyConnections.includes("collections");
+
+  // const includeOrders =
+  //   pluginOptions.shopifyConnections &&
+  //   pluginOptions.shopifyConnections.includes("orders");
+
+  // const productImageFields: Fields = {
+  //   product: {
+  //     type: "ShopifyProduct",
+  //
+  //   }
+  //   //product: `ShopifyProduct @link(from: "productId", by: "id")`,
+  // };
+
+  // if (pluginOptions.downloadImages) {
+  //   productImageFields.localFile = `File @link`;
+  // }
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: "ShopifyProduct",
+      fields: {
+        // collections: includeCollections ? {
+        //   type: "[ShopifyCollection]",
+        //   extensions: {
+        //     directives: [{
+        //       name: "link",
+        //       args: {
+        //         from: "id",
+        //         by: "productIds"
+        //       }
+        //     }]
+        //   }
+        // } : undefined,
+        variants: {
```

Comment:
@vladar I'm wondering if you can see what I'm doing wrong here? I can't get product connections to show up this way. Both `variants` and `images` come back null.